src/main/scala/io/github/sentenza/hacktoberfest18/algos/MutableSorting.scala

```diff
@@ -185,10 +188,44 @@ package io.github.sentenza.hacktoberfest18.algos
       array
     }
 
-    def bucketSort(xs: Array[Int]): Array[Int] = ???
+    def bucketSort(xs: Array[Int]): Array[Int] = bucketSort(xs, sqrt(xs.length).toInt, insertionSort)
     def countSort(xs: Array[Int]): Array[Int] = ???
     def radixSort(xs: Array[Int]): Array[Int] = ???
   
+    /**
+      * Bucket sort is a sorting algorithm that sorts an array of elements by splitting the elements into n buckets
+      * and then reapplies another sorting method on the different buckets. Then merges the sorted buckets.
+      * It can also be used recursively until buckets contain one element each.
+      *
+      * @param array Array of sortable integers
+      * @param n number of buckets in which to sort the elements.
+      * @param sort The sorting algorithm to apply once buckets are divided.
+      * @return the sorted array
+      */
+    def bucketSort(array: Array[Int], n: Int = 10, sort: Array[Int] => Array[Int] = insertionSort): Array[Int] = {
+      //Using the square root of the amount of buckets such that each bucket
+      //will have sqrt(n) items in it on average.
+      val n = sqrt(array.length).toInt
```

Comment:
You redefine `n` using the square root of array dimension. It makes no sense to pass `n` as a parameter. Am I missing anything?